\documentclass{article}[12pt]
\usepackage[margin=0.75in]{geometry}
\usepackage{color}
\usepackage{graphicx}
%\usepackage{hyperref}
\usepackage{fancyvrb}
\usepackage{palatino}
\usepackage{enumitem}
\usepackage{natbib}
\usepackage[T1]{fontenc} % quotes
%\hypersetup{
%	colorlinks=true, %set true if you want colored links
%	linktoc=all,     %set to all if you want both sections and subsections linked
%	linkcolor=blue,  %choose some color if you want links to stand out
%	urlcolor=blue
%}
\newcommand{\code}[1]{\texttt{#1}}  % let's not use bold for now
\sloppypar
\setlength\parindent{0pt}


\begin{document}

\title{Introduction to Python \\ Day Four Exercises}
\author{Stephanie Spielman \\ \footnotesize{Email: stephanie.spielman@gmail.com}}
\date{}
\maketitle{}

\section{Manipulating Sequence Data in BioPython}


\begin{enumerate}[itemsep=5ex]

	\item The first exercise will focus on manipulating sequences in the file "dopamine\_sequences.fasta". Perform the following tasks:
	
	\begin{enumerate}[itemsep=2ex]
		\item Use the SeqIO module to read in the file "dopamine\_sequences.fasta".
		\item Determine the length for each sequence, and save this information to a dictionary. The keys should be the sequence ID, and the values should be the length.
		\item Count the number of guanines per sequence, and again save this information in a dictionary. (Hint: BioPython has implemented certain useful string methods, like \code{.count()} for \code{Seq} object sequences.)
		\item Re-do parts (b) and (c) by writing \emph{functions} to count the sequence length and determine the number of tryptophans. Each function should take a \emph{single} \code{SeqRecord} object as the argument.
		\item Create a new file \textcolor{red}{\textbf{with format XXX}}.
		
		
	\end{enumerate}
	
	\item The first exercise will focus on manipulating sequences in the \emph{sequence alignment} file "some\_alignment.fasta". Perform the following tasks:
		
	\begin{enumerate}[itemsep=2ex]
		\item Read in the file 
		\item Translate the records. These sequences are known to have a motif XYZ at position A. Determine which, if any, of these sequences do not have this motif at that position (hint: remember Python indexing starts from 0, so position A is actually index (A-1)). Print any IDs which do not have this motif.
		\item Save the file in phylip format. 			
			
		\end{enumerate}
	
\end{enumerate}	
	





\bibliographystyle{plainnat}
\bibliography{citations}
















\end{document}
